### ISSUE TITLE
building docker image gives error from wasm-pack in step 19/28

### ISSUE THREAD

**Issue by suppadeliux** [tinysearch/tinysearch#138]
Hello,

I am trying to test the docker but when running  `docker build --build-arg RUST_IMAGE=rustlang/rust:nightly-alpine -t tinysearch/cli:nightly .`

at step 19 from 28, the build stops with the error =>

```bash
Step 19/28 : RUN wasm-pack --version
 ---> Running in d0787575a530
The command '/bin/sh -c wasm-pack --version' returned a non-zero code: 139
```

I tried with others docker images like `docker build --build-arg WASM_BRANCH=master --build-arg TINY_MAGIC=64 -t tinysearch/cli .` or other solutions from this thread => `https://github.com/tinysearch/tinysearch/issues/111`

But it always lead to this error.

Does anyone know why this happened?.

Thanks for the help.

**Comment by johnmathews** [comment]
I have this problem too. Any suggestions would be appreciated.

**Comment by ryanseddon** [comment]
I ended writing my own Dockerfile after having issues with this one:

```Dockerfile
# Docker tinysearch with deps
#   - binaryen
#   - wasm-pack
#   - terser

ARG TINY_REPO=https://github.com/tinysearch/tinysearch
ARG TINY_BRANCH=master
ARG RUST_IMAGE=rust:alpine

FROM $RUST_IMAGE AS binary-build

ARG WASM_REPO
ARG WASM_BRANCH
ARG TINY_REPO
ARG TINY_BRANCH
ARG TINY_MAGIC

WORKDIR /tmp

RUN apk add --update --no-cache --virtual build-dependencies musl-dev openssl-dev gcc curl git npm gcc ca-certificates libc6-compat binaryen

RUN set -eux -o pipefail; \
    ln -s /lib64/ld-linux-x86-64.so.2 /lib/ld64.so.1; \
    npm install terser -g;

RUN cd /tmp && git clone --branch "$TINY_BRANCH" "$TINY_REPO"

RUN curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh

# https://github.com/tinysearch/tinysearch/issues/111
RUN set -ex -o pipefail; cd /tmp/tinysearch && if ! [[ -z $TINY_MAGIC ]]; then sed -i.bak bin/src/storage.rs -e "s/let mut filter = CuckooFilter::with_capacity(words.len() + .*);/let mut filter = CuckooFilter::with_capacity(words.len() + $TINY_MAGIC);/g";fi && cargo build --release && cp target/release/tinysearch $CARGO_HOME/bin && echo $TINY_MAGIC |tee /.tinymagic

RUN wasm-pack --version

FROM $RUST_IMAGE

WORKDIR /tmp

RUN apk add --update --no-cache libc6-compat musl-dev binaryen

RUN set -eux -o pipefail; \
    ln -s /lib64/ld-linux-x86-64.so.2 /lib/ld64.so.1;

COPY --from=binary-build /usr/local/bin/ /usr/local/bin/
COPY --from=binary-build /usr/local/cargo/bin/ /usr/local/bin/
COPY --from=binary-build /usr/bin/terser /usr/local/bin/

# crate cache init. No need to download crate for futur usage
RUN set -eux -o pipefail; \
    echo '[{"title":"","body":"","url":""}]' > build.json; \
    tinysearch build.json; \
    rm -rf /tmp/*

ENTRYPOINT ["tinysearch"]
```

**Comment by johnmathews** [comment]
@ryanseddon would you make a pull request for this?

**Comment by mre** [comment]
Hey y'all. Thanks for your patience.
@johnmathews fixed that in #139. Should be in `master` now. Let me know if you still run into issues. We can reopen the issue then.
Thanks everyone for helping each other out here. Much appreciated.

**Comment by mre** [comment]
Oh yeah, and if anyone is using tinysearch for a project, please create a PR to add a link to the `README.md`. There is a "users" section there. Would love to add a backlink to your project.
